### ISSUE TITLE
Lexicon not addable

### ISSUE THREAD

**Issue by tseaver** [zopefoundation/Zope#149]
Iside a catalog, it doesn't appear in the "Add" menu. But it *is* addable at the root.  ISTM that the most sensible place to put a lexicon is inside a catalog.

**Comment by tseaver** [comment]
Correction:  `ZCTextIndex Lexicon` is a choice in the "Add" menu outside a catalog, but selecting it leads to:

```python
Traceback (most recent call last):
  File "/tmp/z2py36/lib/python3.6/site-packages/waitress/channel.py", line 338, in service
    task.service()
  File "/tmp/z2py36/lib/python3.6/site-packages/waitress/task.py", line 169, in service
    self.execute()
  File "/tmp/z2py36/lib/python3.6/site-packages/waitress/task.py", line 399, in execute
    app_iter = self.channel.server.application(env, start_response)
  File "/tmp/z2py36/lib/python3.6/site-packages/ZPublisher/httpexceptions.py", line 33, in __call__
    return self.application(environ, start_response)
  File "/tmp/z2py36/lib/python3.6/site-packages/ZPublisher/WSGIPublisher.py", line 238, in publish_module
    request, response, module_info, _publish=_publish)
  File "/tmp/z2py36/lib/python3.6/site-packages/ZPublisher/WSGIPublisher.py", line 211, in _publish_response
    reraise(exc.__class__, exc, sys.exc_info()[2])
  File "/tmp/z2py36/lib/python3.6/site-packages/six.py", line 686, in reraise
    raise value
  File "/tmp/z2py36/lib/python3.6/site-packages/ZPublisher/WSGIPublisher.py", line 172, in _publish_response
    response = _publish(request, module_info)
  File "/tmp/z2py36/lib/python3.6/site-packages/ZPublisher/WSGIPublisher.py", line 162, in publish
    bind=1)
  File "/tmp/z2py36/lib/python3.6/site-packages/ZPublisher/mapply.py", line 85, in mapply
    return debug(object, args, context)
  File "/tmp/z2py36/lib/python3.6/site-packages/ZPublisher/WSGIPublisher.py", line 57, in call_object
    return obj(*args)
  File "/tmp/z2py36/lib/python3.6/site-packages/Shared/DC/Scripts/Bindings.py", line 323, in __call__
    return self._bindAndExec(args, kw, None)
  File "/tmp/z2py36/lib/python3.6/site-packages/Shared/DC/Scripts/Bindings.py", line 360, in _bindAndExec
    return self._exec(bound_data, args, kw)
  File "/tmp/z2py36/lib/python3.6/site-packages/App/special_dtml.py", line 203, in _exec
    result = render_blocks(self._v_blocks, ns)
  File "/tmp/z2py36/lib/python3.6/site-packages/DocumentTemplate/_DocumentTemplate.py", line 147, in render_blocks
    render_blocks_(blocks, rendered, md)
  File "/tmp/z2py36/lib/python3.6/site-packages/DocumentTemplate/_DocumentTemplate.py", line 248, in render_blocks_
    block = block(md)
  File "/tmp/z2py36/lib/python3.6/site-packages/DocumentTemplate/DT_In.py", line 700, in renderwob
    for k, v in kw.items():
RuntimeError: dictionary changed size during iteration
```

**Comment by tseaver** [comment]
This traceback is really https://github.com/zopefoundation/DocumentTemplate/issues/10.

We still need to fix the add menu issue, however.

**Comment by sallner** [comment]
@tseaver I failed to reproduce your issue, but maybe I misunderstood. In a fresh instance of Zope4, I added a `ZCatalog` to the root and then it was possible to add a `ZCTextIndex Lexicon` in the root folder and in the ZCatalog. It is not possible to add it on the `Indexes`-Tab, but that was not possible in Zope 2.13 either.  

Can you clarify that issue please?

**Comment by dataflake** [comment]
IMHO this can be closed. The actual issue causing the traceback is fixed in DocumentTemplate since 3.0a4 (Zope now requires 3.0b1) and the behavior that you can add Lexicon instances outside a catalog is unchanged from prior Zope versions.
